packages/gatsby-plugin-fastify/src/utils/headers.ts

```diff
@@ -38,3 +47,34 @@ export function moduleHeaderDecorator(this: FastifyReply, module: Modules): void
 export function setHeaderDecorator(this: FastifyReply, key: string, value: string) {
   this.header(key, value);
 }
+
+export function appendRouteHeaders(route: string, reply: FastifyReply): Record<string, string> {
+  const {
+    server: { headers },
+  } = getConfig();
+
+  /**
+   * TODO: Solve This
+   * If the route is a match for a header, return the header value
+   * But how do we prioritize certain matches.
+   * How do I make sure that the header is not overwritten by other matches?
+   * we expect this but we need a way to prioritize one match over another
+   * or am I over complicating this?
+   *
+   * @tsdexter: I think we can just use the order of the headers in the
+   * config file. Successive matches will with duplicate header `name`'s
+   * will overwrite the previous header of the same `name`.
+   */
+  // console.log(`\n\nheaders`, {headers}, `\n\n`);
+  let routeHeaders = {};
+  // console.log(`\n\nheaders`, {headers}, `\n\n`);
+  mapObject(headers, (pattern, headerObject) => {
+    if (isMatch(route, pattern)) {
```

Comment:
We should include match examples in docs (I believe you did this), but should also include a note that we use `picomatch` and link to those complete docs.

Reply:
Dope